builtin/rebase.c

```diff
@@ -34,8 +41,97 @@ static int use_builtin_rebase(void)
 	return ret;
 }
 
+static int apply_autostash(void)
+{
+	warning("TODO");
+	return 0;
+}
+
+struct rebase_options {
+	enum rebase_type type;
+	int preserve_merges;
+	const char *upstream_name;
+	const char *state_dir;
+};
+
+static int finish_rebase(struct rebase_options *opts)
+{
+	struct strbuf dir = STRBUF_INIT;
+	const char *argv_gc_auto[] = { "gc", "--auto", NULL };
+
+	delete_ref(NULL, "REBASE_HEAD", NULL, REF_NO_DEREF);
+	apply_autostash();
+	/*
+	 * We ignore errors in 'gc --auto', since the
+	 * user should see them.
+	 */
+	close_all_packs(the_repository->objects);
+	run_command_v_opt(argv_gc_auto, RUN_GIT_CMD);
+	strbuf_addstr(&dir, opts->state_dir);
+	remove_dir_recursively(&dir, 0);
+	strbuf_release(&dir);
+
+	return 0;
+}
+
+static int run_specific_rebase(struct rebase_options *opts)
+{
+	const char *argv[] = { NULL, NULL };
```

Comment:
It is actually not at all a good use case for the argv array because we have to call a Unix shell script snippet here: the rebase backends are not stand-alone scripts, but instead they provide *Unix shell script functions* that need to be called after sourcing the backend files.

What's more: these backends use Unix shell functions defined both by `git-sh-setup.sh` and `git-rebase.sh` (we move the latter's into `git-rebase--common.sh` to accommodate for that), so we not only have to source the backend file before calling the respective Unix shell script function, but we have to source `git-sh-setup` and `git-rebase--common` before that.

And since this is all done in a Unix shell script snippet, all of this is in `argv[0]`. There never will be a non-NULL `argv[1]`. So in this case, using a `struct argv_array` would be particularly pointless.

Reply:
@prertik maybe you can add this explanation to the commit message so that future reviewers are less likely to stumble over this non-use of `argv_array`?